File: software/examples/c/example_simple.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_multi_touch.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

int main() {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	MultiTouch mt;
	multi_touch_create(&mt, UID, &ipcon); 

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		exit(1);
	}
	// Don't use device before ipcon is connected

	// Get current touch_state
	uint16_t touch_state;
	if(multi_touch_get_touch_state(&mt, &touch_state) < 0) {
		fprintf(stderr, "Could not get value, probably timeout\n");
		exit(1);
	}

	if(touch_state & (1 << 12)) {
		printf("In proximity, ");
	}

	if((touch_state & 0xfff) == 0) {
		printf("No electrodes touched\n\n");
	} else {
		printf("Electrodes ");
		int i = 0;
		for(; i < 12; i++) {
			if(touch_state & (1 << i)) {
				printf("%d ", i);
			}
		}
		printf("touched\n\n");
	}

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
    "github.com/tinkerforge/go-api-bindings/ipconnection"
    "github.com/tinkerforge/go-api-bindings/multi_touch_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Multi Touch Bricklet.

func main() {
	ipcon := ipconnection.New()
    defer ipcon.Close()
	mt, _ := multi_touch_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
    defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current touch state.
	state, _ := mt.GetTouchState()
    var str string
    
    if (state & (1 << 12)) == (1 << 12) {
        str += "In proximity, "
    }
    
    if (state & 0xfff) == 0 {
        str += "No electrodes touched"
    } else {
        str += "Electrodes "
        for i := uint(0); i < 12; i++ {
            if (state & (1 << i)) == (1 << i) {
                str += string(i) + " "
            }
        }
        str += "touched"
    }    
    
	fmt.Print("Press enter to exit.")
	fmt.Scanln()

	ipcon.Disconnect()
}
